lib/host/wasi/wasifunc.cpp

```diff
@@ -1512,18 +1519,58 @@ Expect<uint32_t> WasiPollOneoff<Trigger>::body(
       }
       case __WASI_EVENTTYPE_FD_READ: {
         const __wasi_fd_t WasiFd = Sub.u.u.fd_read.file_descriptor;
-        Poller.read(WasiFd, Trigger, WasiUserData);
+        auto Iter = SubInfoMap.find(WasiFd);
+        const bool NotFound = Iter == SubInfoMap.end();
+        if (NotFound) {
+          SubInfo Info = {.ReadFlag = true,
```

Comment:
Don't use this pattern. It's not in the C++17 standard. It's c99 standard, and will be shipped in C++20 (ref: https://www.cppstories.com/2021/designated-init-cpp20/)